src/coreclr/vm/riscv64/profiler.cpp

```diff
@@ -119,17 +90,77 @@ ProfileArgIterator::ProfileArgIterator(MetaSig* pSig, void* pPlatformSpecificHan
 
 ProfileArgIterator::~ProfileArgIterator()
 {
-    // TODO-RISCV64-CQ: copied codes from loongarch however not yet tested.
-    _ASSERTE(!"RISCV64:NYI");
     LIMITED_METHOD_CONTRACT;
 
     m_handle = nullptr;
 }
 
+LPVOID ProfileArgIterator::CopyStructFromRegisters(const ArgLocDesc* sir)
+{
+    struct Func
+    {
+        static inline const BYTE* postIncrement(const BYTE *&p, int offset)
+        {
+            const BYTE* orig = p;
+            p += offset;
+            return orig;
+        }
+    };
+
+    WRAPPER_NO_CONTRACT;
+    _ASSERTE(m_handle);
+    PROFILE_PLATFORM_SPECIFIC_DATA* pData = reinterpret_cast<PROFILE_PLATFORM_SPECIFIC_DATA*>(m_handle);
+
+    struct { bool isFloat, is8; } fields[] = {
+        { sir->m_structFields & (STRUCT_FLOAT_FIELD_FIRST | STRUCT_FLOAT_FIELD_ONLY_TWO | STRUCT_FLOAT_FIELD_ONLY_ONE),
+          sir->m_structFields & STRUCT_FIRST_FIELD_SIZE_IS8 },
+        { sir->m_structFields & (STRUCT_FLOAT_FIELD_SECOND | STRUCT_FLOAT_FIELD_ONLY_TWO),
+          sir->m_structFields & STRUCT_SECOND_FIELD_SIZE_IS8 },
+    };
+    int fieldCount = (sir->m_structFields & STRUCT_FLOAT_FIELD_ONLY_ONE) ? 1 : 2;
+    UINT64 bufferPosBegin = m_bufferPos;
+    const double *fRegBegin = &pData->floatArgumentRegisters.f[sir->m_idxFloatReg], *fReg = fRegBegin;
+    const double *fRegEnd = fReg + sizeof(pData->floatArgumentRegisters.f)/sizeof(pData->floatArgumentRegisters.f[0]);
+    const INT64 *aRegBegin = &pData->argumentRegisters.a[sir->m_idxGenReg], *aReg = aRegBegin;
+    const INT64 *aRegEnd = aReg + sizeof(pData->argumentRegisters.a)/sizeof(pData->argumentRegisters.a[0]);
```

Comment:
I think you can use `NUM_ARGUMENT_REGISTERS` and `NUM_FLOAT_ARGUMENT_REGISTERS`.
And updates `ArgumentRegisters` and `FloatArgumentRegisters` structures to use the definitions instead of number `8`.

Reply:
OK